couchbase/retry_reason.hxx

```diff
@@ -99,6 +101,9 @@ enum class retry_reason {
     views_no_active_partition,
 };
 
+auto
+retry_reason_to_enum(const std::string& reason) -> couchbase::retry_reason;
```

Comment:
Do we need to expose this in the Public API?

Reply:
no